core/includes/install.core.inc

```diff
@@ -1106,8 +1106,9 @@ function install_settings_form_submit($form, &$form_state) {
   );
 
   // If UTF-8 MB4 is not supported, use the old UTF-8 charset.
-  if (install_verify_database_utf8mb4() === FALSE && $db_settings['charset'] === 'utf8mb4') {
+  if (install_verify_database_utf8mb4() === FALSE) {
     $db_settings['charset'] = 'utf8';
+    $db_settings['collation'] = 'utf8_general_ci';
```

Comment:
Before, collation wasn't set so the database would use its default.

```suggestion
    $db_settings['collation'] = '';
```

On the following line:
```php
  if ($db_settings['collation'] && $db_settings['collation'] != 'utf8mb4_general_ci') {
    $settings['database']['value']['collation'] = $db_settings['collation'];
  }
```

Change to:

```php
  if (isset($db_settings['collation']) && $db_settings['collation'] != 'utf8mb4_general_ci') {
    $settings['database']['value']['collation'] = $db_settings['collation'];
  }
```

Reply:
> Before, collation wasn't set so the database would use its default.

Sure, let's try that instead.